includes/core/class-wc-payments-order.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Class file for WCPay\Core\WC_Payments_Order
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WC_Order;
+use WC_Payments_Utils;
+
+/**
+ * WC_Payments_Order
+ */
+class WC_Payments_Order {
+
+	/**
+	 * The WC_Order corresponding to this WC_Payments_Order.
+	 *
+	 * @var WC_Order
+	 */
+	private $order;
+
+	/**
+	 * Constructor when order_id is provided.
+	 *
+	 * @param WC_Order|int $order The order Id.
+	 */
+	public function __construct( $order ) {
+		$this->order = wc_get_order( $order );
+	}
+
+	/**
+	 * Get the payment metadata for key _intent_id.
+	 *
+	 * @return string
+	 */
+	public function get_payment_intent_id() : string {
+		return $this->order->get_meta( '_intent_id', true );
```

Comment:
This could be a great opportunity to introduce stronger types. I'd check if there is a non-empty string and return `string`, otherwise `null`.